tests/testthat/test_missingLevelsTrain.R

```diff
@@ -0,0 +1,38 @@
+context("missingLevelsTrain.R")
+
+test_that("additional factor levels in test data are removed and resample() does not error", {
+
+  task = sonar.task
+  # introduce factor level with only one observation
+  task$env$data$test_fac = factor(c(rep("1", 1), c(rep("2", 3),
+    rep("3", length(task$env$data$V1) - 4))))
```

Comment:
Note that the data in `task$env` has reference semantics: You are actually *changing* the `sonar.task` here, which might introduce hard to fix bugs in other (future?) tests. I would do `task = subsetTask(sonar.task)`, which creates a copy.

Reply:
Ok, interesting. Did not know that because I thought that I just change the `task` object which got created from `sonar.task`. Nevertheless, I changed it to your suggestions if that is more robust :+1: